### PR TITLE
[ADZ-1365] API catalogue categories restyle

### PR DESCRIPTION
**Pull request by ziemsky**
Category headings in API catalogue's _Filters_
section are now styled in line with the Design
System > Filter Menu Section.

Also, implements [ADZ-1376] '_API Catalogue -
Side Nav Refinement (Filters to remain
expanded when selected)_'.

### CODE REVIEW

File: acceptance-tests/pom.xml [hunk 1]
```diff
@@ -20,7 +20,11 @@
         <cucumber.version>1.2.5</cucumber.version>
         <spring.version>5.1.15.RELEASE</spring.version>
         <selenium.version>3.141.59</selenium.version>
-        <chromedriver.version>92.0.4515.43</chromedriver.version>
+        <!--
+        Consult https://chromedriver.chromium.org/downloads
+        to find the right ChromeDriver version to use with the current browser version,
+        -->
+        <chromedriver.version>93.0.4577.63</chromedriver.version>
```
Comment (harryhmc) [inline acceptance-tests/pom.xml:27]: out of curiosity, what was this version update required for? :smile:

File: repository-data/webfiles/src/main/resources/site/src/scss/rebrand/components/_api-catalogue.scss [hunk 2]
```diff
@@ -50,11 +50,13 @@
     }
   }
 
-  input.section-folder:not(:checked) {
-
-    ~ label .icon svg {
-      rotate: -90deg;
+  input.section-folder:checked {
+    ~ .nhsd-m-filter-menu-section__menu-button label .icon svg {
```
Comment (harryhmc) [inline repository-data/webfiles/src/main/resources/site/src/scss/rebrand/components/_api-catalogue.scss:54]: i would just write `~.nhsd-m-filter-menu-section__menu-button svg` here, it isn't necessary to target each subsequent child when we only have one svg in the div anyway

File: repository-data/webfiles/src/main/resources/site/src/scss/rebrand/components/_api-catalogue.scss [hunk 3]
```diff
@@ -50,11 +50,13 @@
...
+      rotate: 180deg;
     }
+  }
+
+  input.section-folder:not(:checked) {
```
Comment (harryhmc) [inline repository-data/webfiles/src/main/resources/site/src/scss/rebrand/components/_api-catalogue.scss:59]: these can be written in a single `input.section-folder {...}` block by using `&:checked`, `&:not(:checked)` as your nested selectors

File: repository-data/webfiles/src/main/resources/site/src/scss/rebrand/components/_api-catalogue.scss [hunk 4]
```diff
@@ -50,11 +50,13 @@
...
 
     ~ .section-entries {
```
Comment (harryhmc) [inline repository-data/webfiles/src/main/resources/site/src/scss/rebrand/components/_api-catalogue.scss:61]: not your code, but could you combine this with the `hr:first-of-type` rule below?

File: repository-data/webfiles/src/main/resources/site/freemarker/common/apicatalogue/scrollable-filter-nav.ftl [hunk 5]
```diff
@@ -32,20 +32,28 @@
...
+                            <input id="<#if responsive>responsive_</#if>toggler_${section.key}"
+                                   class="section-folder"
+                                   style="display:none;"
+                                   type="checkbox"
+                                   aria-hidden="true"
+                                   <#if section.expanded>checked</#if>/>
+                            <div class="nhsd-m-filter-menu-section__menu-button">
+                                <label for="<#if responsive>responsive_</#if>toggler_${section.key}"
+                                       class="nhsd-m-filter-menu-section__heading-text">
+                                    <span class="icon">
```
Comment (harryhmc) [inline repository-data/webfiles/src/main/resources/site/freemarker/common/apicatalogue/scrollable-filter-nav.ftl:45]: not your code -- but do we need this extra <span> here?

File: acceptance-tests/pom.xml (hunk 1 above)
Comment (ziemsky) [reply acceptance-tests/pom.xml:27]: ChromeDriver is used by Selenium test framework as part of automated acceptance tests, targeting headless Chrome web browser.

Versions of ChromeDriver roughly track versions of Chrome, and when the latter gets updated (especially if it's a major version number bump) ChromeDriver needs updating to corresponding as well.

In this case, Chrome available on GitHub has moved forward and the build started failing as a result.

File: repository-data/webfiles/src/main/resources/site/src/scss/rebrand/components/_api-catalogue.scss (hunk 2 above)
Comment (ziemsky) [reply repository-data/webfiles/src/main/resources/site/src/scss/rebrand/components/_api-catalogue.scss:54]: Indeed. Fixing.

File: repository-data/webfiles/src/main/resources/site/src/scss/rebrand/components/_api-catalogue.scss (hunk 3 above)
Comment (ziemsky) [reply repository-data/webfiles/src/main/resources/site/src/scss/rebrand/components/_api-catalogue.scss:59]: Another good spot! Fixing now.

File: repository-data/webfiles/src/main/resources/site/src/scss/rebrand/components/_api-catalogue.scss (hunk 4 above)
Comment (ziemsky) [reply repository-data/webfiles/src/main/resources/site/src/scss/rebrand/components/_api-catalogue.scss:61]: Doing it now.

File: repository-data/webfiles/src/main/resources/site/freemarker/common/apicatalogue/scrollable-filter-nav.ftl (hunk 5 above)
Comment (ziemsky) [reply repository-data/webfiles/src/main/resources/site/freemarker/common/apicatalogue/scrollable-filter-nav.ftl:45]: Now that we have eliminated the reference to class 'icon' from the SCSS file - we don't need this element here.

I'm guessing it's presence here was inspired by the Design Catalogue, but it's redundant in our context.

Now removed.

**Review by harryhmc — APPROVED** [review]
LGTM, great work :+1:

**Comment by ziemsky** [comment]
`7be97e3` is a squash of the two reviewed commits.

**Pull request merged**